src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -72,21 +68,28 @@ public boolean isOwner(NsUser loginUser) {
         return writer.equals(loginUser);
     }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
-
     public boolean isDeleted() {
         return deleted;
     }
 
-    public List<Answer> getAnswers() {
-        return answers;
-    }
-
     @Override
     public String toString() {
         return "Question [id=" + getId() + ", title=" + title + ", contents=" + contents + ", writer=" + writer + "]";
     }
-}
+
+    public List<DeleteHistory> delete(NsUser user) {
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
+
+        questionUserCheck(user);
+        this.deleted = true;
+        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, id, writer));
+        deleteHistories.addAll(answers.deleteAnswers(user));
+        return deleteHistories;
```

Comment:
보통 함수를 설계할 때 함수의 추상화 레벨을 동일하게 유지하라고 권장드립니다. 
위 함수들도 다음과같이 추상화 레벨을 맞춰볼 수 있을 것 같네요.

```suggestion
        if(!deletable(user)) {
            // 삭제 불가능시 로직
       }
        List<DeleteHistory> deleteHistories = List.of(deleteQuestion());
        deleteHistories.addAll(deleteAnswers());
        
        return deleteHistories;
```